src/build.d

```diff
@@ -991,21 +991,29 @@ void processEnvironment()
 
     const os = env["OS"];
 
-    const hostDMDVersion = [env["HOST_DMD_RUN"], "--version"].execute.output;
+    // Detect the host compiler kind and version
+    const hostDmdInfo = [env["HOST_DMD_RUN"], `-Xi=compilerInfo`, `-Xf=-`].execute();
+    writeln(hostDmdInfo);
 
-    alias DMD = AliasSeq!("DMD");
-    alias LDC = AliasSeq!("LDC");
-    alias GDC = AliasSeq!("GDC", "gdmd", "gdc");
-    const kindIdx = hostDMDVersion.canFind(DMD, LDC, GDC);
-
-    enforce(kindIdx, "Invalid Host DMD found: " ~ hostDMDVersion);
-
-    if (kindIdx <= DMD.length)
-        env["HOST_DMD_KIND"] = "dmd";
-    else if (kindIdx <= LDC.length + DMD.length)
-        env["HOST_DMD_KIND"] = "ldc";
-    else
+    if (hostDmdInfo.status) // Failed, JSON output currently not supported for GDC
+    {
         env["HOST_DMD_KIND"] = "gdc";
+        env["HOST_DMD_VERSION"] = "v2.079";
+    }
+    else
+    {
+        // Reads the content of a single field without parsing the entire JSON
+        alias get = field => hostDmdInfo.output
+            .findSplitAfter(field ~ `" : "`)[1]
+            .findSplitBefore(`"`)[0];
+
+        env["HOST_DMD_KIND"] = [
+            "Digital Mars D": "dmd",
+            "LDC": "ldc",
+            "GNU D": "gdc"
+        ][get(`vendor`)];
+        env["HOST_DMD_VERSION"] = get(`version`)[1 .. "vX.XXX.X".length];
```

Comment:
So apparently in the conversion to D strings it was unnoticed that `_version` gets converted into a null-terminated string.

Details: https://github.com/dlang/dmd/pull/11640

Reply:
Yeah, `"version" : "v2.093.1-dirty\u0000"` isn't right.